src/main/kotlin/io/hsar/wh40k/combatsimulator/logic/TacticalActionStrategy.kt

```diff
@@ -2,34 +2,68 @@ package io.hsar.wh40k.combatsimulator.logic
 
 import io.hsar.wh40k.combatsimulator.model.UnitInstance
 import io.hsar.wh40k.combatsimulator.model.World
+import io.hsar.wh40k.combatsimulator.model.unit.BaseStat
 import io.hsar.wh40k.combatsimulator.random.AverageDice
 
 object TacticalActionStrategy : ActionStrategy {
 
-    override fun decideTurnActions(world: World, thisUnit: UnitInstance, possibleActions: Collection<TurnAction>): List<TurnAction> {
+    override fun decideTurnActions(world: World, thisUnit: UnitInstance, possibleActionOptions: Collection<ActionOption>): List<TurnAction> {
         // Shit implementation - units will only ever aim and fire their max damage attack
-        val aimAction = possibleActions
+        val aimAction = possibleActionOptions
                 .find { action ->
-                    action is TurnAction.HalfAim
+                    action is ActionOption.HalfAim
+                }
+                ?.let { action ->
+                    AimAction(action)
```

Comment:
I'm sure that there's a better way... I'm not sure. Can HalfAim be both an ActionOption and a TurnAction (I've not checked)?

Reply:
Inherently not, as an ActionOption represents the generic option to do that thing whereas a TurnAction contains (at least) an ActionOption and represents a concrete action a unit is taking on a given turn.

Reply:
Shame.